core/TypeDrivenAutocorrect.cc

```diff
@@ -18,6 +18,15 @@ void TypeDrivenAutocorrect::maybeAutocorrect(const GlobalState &gs, ErrorBuilder
         if (!withoutNil.isBottom() &&
             Types::isSubTypeUnderConstraint(gs, constr, withoutNil, expectedType, UntypedMode::AlwaysCompatible)) {
             e.replaceWith("Wrap in `T.must`", loc, "T.must({})", loc.source(gs).value());
+        } else if (Types::isSubTypeUnderConstraint(gs, constr, expectedType, Types::Boolean(),
+                                                   UntypedMode::AlwaysCompatible)) {
+            if (core::isa_type<ClassType>(actualType)) {
+                auto classSymbol = core::cast_type_nonnull<ClassType>(actualType).symbol;
+                if (classSymbol.exists() && classSymbol.data(gs)->owner == core::Symbols::root() &&
+                    classSymbol.data(gs)->name == core::Names::Constants::Boolean()) {
```

Comment:
Do we want to suggest in the case where `actualType` is `::TrueClass` or `::FalseClass`? (Or is that not possible for some reason?)

Reply:
`TrueClass` is already a subtype of `T::Boolean`, so it wouldn't report an error and we wouldn't get here.

Reply:
Oh yeah, good point :)